### PR TITLE
refactor(trading): remove dead confirm page plumbing

### LINKED ISSUE #29388: Cleanup after removing context from all confirm pages

### PR DESCRIPTION
**Pull request by sherpaPSX**
## Description

- Remove dead `pageType`, confirm, and retry plumbing from trading form hooks and initializer now that these hooks only back the form pages.
- Resolve buy, sell, and exchange detail providers from trading selectors instead of the deleted provider lookup helper.
- Simplify `TradingContainer` to render children directly and inline the trading form layouts that use it.
- Delete unused trading UI helpers and route-level leftovers that became dead after the confirm page cleanup.

## Notes for QA

Just dev thing

## Related Issue

Resolve #29388

## Screenshots:

N/A

<details>
<summary><h3>🤖 LLM Test Recommendations</h3></summary>

**Summary:** This change set is focused entirely on the trading/coinmarket feature area — buy, sell, and swap (exchange) forms, hooks, views, types, and utilities. The risk is concentrated in trading flows: form initialization, compose transaction, quote handling, receive address selection, and UI rendering for all three trading modes. The recommended strategy is to run all trading-specific E2E tests at high priority since they directly exercise the changed components, with swap input/fee/history and live tests at medium priority. Non-trading tests (analytics, backup, onboarding, etc.) should be skipped as the shared dependencies (tradingUtils, form hooks) are imported transitively but not exercised in those test paths.

<details><summary><b>Changed files (30)</b></summary>

- `packages/suite/src/hooks/wallet/trading/form/common/__tests__/useTradingClearStaleQuotes.test.tsx`
- `packages/suite/src/hooks/wallet/trading/form/common/useTradingClearStaleQuotes.ts`
- `packages/suite/src/hooks/wallet/trading/form/common/useTradingComposeTransaction.ts`
- `packages/suite/src/hooks/wallet/trading/form/common/useTradingFormActions.ts`
- `packages/suite/src/hooks/wallet/trading/form/common/useTradingInitializer.ts`
- `packages/suite/src/hooks/wallet/trading/form/useTradingBuyForm.ts`
- `packages/suite/src/hooks/wallet/trading/form/useTradingExchangeForm.ts`
- `packages/suite/src/hooks/wallet/trading/form/useTradingExchangeWatchApproval.ts`
- `packages/suite/src/hooks/wallet/trading/form/useTradingReceiveAddress.ts`
- `packages/suite/src/hooks/wallet/trading/form/useTradingSellForm.ts`
- `packages/suite/src/types/trading/trading.ts`
- `packages/suite/src/types/trading/tradingForm.ts`
- `packages/suite/src/utils/wallet/trading/__fixtures__/tradingUtils.ts`
- `packages/suite/src/utils/wallet/trading/__tests__/tradingUtils.test.ts`
- `packages/suite/src/utils/wallet/trading/tradingTypingUtils.ts`
- `packages/suite/src/utils/wallet/trading/tradingUtils.ts`
- `packages/suite/src/views/wallet/trading/buy/TradingBuyDetail.tsx`
- `packages/suite/src/views/wallet/trading/buy/TradingBuyForm.tsx`
- `packages/suite/src/views/wallet/trading/common/TradingContainer.tsx`
- `packages/suite/src/views/wallet/trading/common/TradingForm/TradingFormInput/TradingFormInputAssetPicker/AssetPickerInput/AssetPickerAccountLabel.tsx`
- `packages/suite/src/views/wallet/trading/common/TradingSelectedOffer/TradingInfo/TradingInfoHeader.tsx`
- `packages/suite/src/views/wallet/trading/common/TradingUtils/TradingUtilsPrice.tsx`
- `packages/suite/src/views/wallet/trading/common/TradingUtils/TradingUtilsTooltip.tsx`
- `packages/suite/src/views/wallet/trading/common/TradingUtils/TradingUtilsTooltipFee.tsx`
- `packages/suite/src/views/wallet/trading/exchange/TradingExchangeDetail.tsx`
- `packages/suite/src/views/wallet/trading/exchange/TradingExchangeForm.tsx`
- `packages/suite/src/views/wallet/trading/index.tsx`
- `packages/suite/src/views/wallet/trading/sell/TradingSellDetail.tsx`
- `packages/suite/src/views/wallet/trading/sell/TradingSellForm.tsx`
- `packages/suite/src/views/wallet/trading/transactions/TradingTransactions.tsx`

</details>

**Recommended tests (22)**

<details><summary>🔴 <b>High priority (14)</b></summary>

- `suite/e2e/tests/trading/buy-bitcoin.test.ts` — Directly exercises TradingBuyForm, TradingBuyDetail, TradingContainer, TradingUtilsPrice, useTradingBuyForm, and the trading index — all changed. Tests the full buy flow including quotes, confirmation, trade payload, and transaction detail status.
- `suite/e2e/tests/trading/buy-solana-token.test.ts` — Tests buying a Solana SPL token, directly using TradingBuyForm, TradingBuyDetail, useTradingBuyForm, and TradingContainer. Covers token-specific asset picker and receive account flows which touch the changed form hooks and views.
- `suite/e2e/tests/trading/sell-solana.test.ts` — Directly exercises TradingSellForm, TradingSellDetail, TradingUtilsPrice, TradingContainer, useTradingSellForm, and trading index. Tests the full sell flow including form fill, quotes, device confirmation, toast, and transaction status polling.
- `suite/e2e/tests/trading/swap-coins.test.ts` — Tests SOL-to-BTC swap flow exercising TradingExchangeForm, TradingExchangeDetail, TradingContainer, useTradingExchangeForm, and trading index. Covers the full swap lifecycle including quotes, device confirmation, toast, and status polling through multiple states.
- `suite/e2e/tests/trading/sell-bitcoin.test.ts` — Tests the full BTC sell flow through TradingSellForm (implicitly via sell-inputs path), TradingContainer, trading index, and useTradingSellForm. Verifies trade payload, confirmation screen, and device prompt for BTC-specific sell.
- `suite/e2e/tests/trading/sell-inputs.test.ts` — Validates sell form input behavior (decimal validation, insufficient funds, percentage buttons, max button, fiat switching) which directly exercises TradingSellForm, TradingBuyForm, useTradingSellForm, useTradingBuyForm, and TradingContainer.
- `suite/e2e/tests/trading/swap-coin-to-token.test.ts` — Tests SOL-to-USDC swap directly exercising TradingExchangeForm, TradingExchangeDetail, TradingContainer, useTradingExchangeForm, useTradingReceiveAddress, and trading index. Covers cross-network receive account selection.
- `suite/e2e/tests/trading/buy-negative.test.ts` — Tests buy form validation edge cases (max/min limits, empty quotes) which directly use TradingBuyForm, useTradingBuyForm, and TradingContainer. Critical for verifying form validation logic in changed hooks.
- `suite/e2e/tests/trading/buy-ethereum.test.ts` — Tests buying ETH via asset picker and receive account selection, directly exercising TradingBuyForm, TradingBuyDetail, TradingContainer, useTradingBuyForm, useTradingReceiveAddress, and trading index.
- `suite/e2e/tests/trading/navigation.test.ts` — Tests navigation entry points into Buy/Sell/Swap forms from multiple locations, directly exercising TradingBuyForm, TradingSellForm, TradingExchangeForm, and TradingContainer. Verifies forms open correctly with the right preselected asset.
- `suite/e2e/tests/trading/sell-ethereum.test.ts` — Tests selling ETH with custom gas fees, directly exercising TradingContainer, trading index, useTradingSellForm, and the sell confirmation flow. Verifies fee display and confirmation screen.
- `suite/e2e/tests/trading/sell-ethereum-token.test.ts` — Tests selling an ERC-20 token (USDC) exercising TradingContainer, trading index, useTradingSellForm, and tradingTypingUtils. Covers token-specific sell flows.
- `suite/e2e/tests/trading/swap-token-to-coin.test.ts` — Tests swapping USDT token to BTC, directly exercising TradingExchangeForm, TradingExchangeDetail, TradingContainer, useTradingExchangeForm, and trading index. Covers token-to-coin swap path.
- `suite/e2e/tests/trading/swap-tokens.test.ts` — Tests swapping USDT to USDC (token-to-token), directly exercising TradingExchangeForm, TradingExchangeDetail, TradingContainer, useTradingExchangeForm, and trading index.

</details>

<details><summary>🟡 <b>Medium priority (8)</b></summary>

- `suite/e2e/tests/trading/swap-fees-bitcoin.test.ts` — Tests BTC swap with custom fee settings, exercising TradingExchangeForm, TradingContainer, useTradingExchangeForm, and trading index. Fee calculation changes in useTradingComposeTransaction could affect this test.
- `suite/e2e/tests/trading/swap-fees-ethereum.test.ts` — Tests ETH swap with custom gas fees, exercising TradingExchangeForm, TradingContainer, useTradingExchangeForm, and trading index. Custom fee UI depends on compose transaction logic.
- `suite/e2e/tests/trading/swap-token-to-disabled-network.test.ts` — Tests swap to a disabled-network asset, exercising TradingExchangeForm and TradingContainer. A more niche path but directly touches changed form/container components.
- `suite/e2e/tests/trading/swap-inputs.test.ts` — Tests swap form input behavior (fraction buttons, validation, provider selection) with a remembered wallet. Uses TradingExchangeForm, TradingContainer, and tradingTypingUtils. Changes to form hooks could affect input validation.
- `suite/e2e/tests/trading/swap-history.test.ts` — Tests swap trade history display via TradingTransactions.tsx (changed but no static mapping). The transaction list, detail sidebar, and status display could be affected by changes to the transactions view component.
- `suite/e2e/tests/trading-live/live-swap-coin-to-token.test.ts` — Live swap test exercising TradingExchangeForm, TradingExchangeDetail, and TradingContainer with real providers. Changes to exchange form hooks or container could break the live swap flow.
- `suite/e2e/tests/trading-live/live-swap-spl-token-to-coin.test.ts` — Live swap test for SPL token to coin, directly using TradingExchangeForm, TradingExchangeDetail, and TradingContainer. Changes to these components could affect live trading flows.
- `suite/e2e/tests/dashboard/assets.test.ts` — Tests buy action from dashboard assets which navigates to trading page, exercising TradingBuyForm and TradingContainer. Changes to TradingContainer or buy form could affect the navigation-to-trading flow.

</details>

⚠️ **Changes with no test coverage (11)**
- `packages/suite/src/hooks/wallet/trading/form/common/__tests__/useTradingClearStaleQuotes.test.tsx`
- `packages/suite/src/hooks/wallet/trading/form/common/useTradingClearStaleQuotes.ts`
- `packages/suite/src/hooks/wallet/trading/form/useTradingExchangeWatchApproval.ts`
- `packages/suite/src/types/trading/trading.ts`
- `packages/suite/src/types/trading/tradingForm.ts`
- `packages/suite/src/utils/wallet/trading/__fixtures__/tradingUtils.ts`
- `packages/suite/src/utils/wallet/trading/__tests__/tradingUtils.test.ts`
- `packages/suite/src/views/wallet/trading/common/TradingForm/TradingFormInput/TradingFormInputAssetPicker/AssetPickerInput/AssetPickerAccountLabel.tsx`
- `packages/suite/src/views/wallet/trading/common/TradingSelectedOffer/TradingInfo/TradingInfoHeader.tsx`
- `packages/suite/src/views/wallet/trading/common/TradingUtils/TradingUtilsTooltip.tsx`
- `packages/suite/src/views/wallet/trading/common/TradingUtils/TradingUtilsTooltipFee.tsx`

_Updated: 2026-07-08T13:42:11.172Z_

</details>

## 🔍 Currents Test Results

🔍 **Suite desktop test results:** [View in Currents](https://app.currents.dev/projects/4ytF0E/runs?branch=feat/29388-trading-confirm-cleanup&lookback=7)

🔍 **Suite web test results:** [View in Currents](https://app.currents.dev/projects/Og0NOQ/runs?branch=feat/29388-trading-confirm-cleanup&lookback=7)

## 🔒 Quarantined E2E Tests

<details><summary>Trezor Suite (web) — 1 test(s)</summary>

| Test | Type |
|------|------|
| Quarantine test: "TrezorConnect webextension -> Suite Web,second call after popup was closed by user should work" | 🙋 manual |

</details>

_Updated: 2026-07-09T06:13:15.825Z • 1 test(s) total_

<details><summary>Trezor Suite (desktop) — 0 test(s)</summary>

_No quarantined tests_ ✅

</details>

_Updated: 2026-07-09T06:05:52.826Z • 0 test(s) total_

## 🌐 Preview deployments

🌐 **Suite Web preview:** [https://dev.suite.sldev.cz/suite-web/feat/29388-trading-confirm-cleanup/web/](https://dev.suite.sldev.cz/suite-web/feat/29388-trading-confirm-cleanup/web/)

### CODE REVIEW

**Comment by sherpaPSX** [comment]
@coderabbit review

File: packages/suite/src/hooks/wallet/trading/form/common/useTradingComposeTransaction.ts [hunk 1]
```diff
@@ -78,8 +77,7 @@ export const useTradingComposeTransaction = <T extends TradingSellExchangeFormPr
 
     useEffect(() => {
         const currentDevice = deviceRef.current;
-        // deriving during sign/review would fire a device call that tears down the modal
-        if (!estimateFeeForUnknownRecipient || networkType !== 'tron' || !currentDevice) {
+        if (networkType !== 'tron' || !currentDevice) {
```
Comment (Lemonexe) [inline packages/suite/src/hooks/wallet/trading/form/common/useTradingComposeTransaction.ts:80]: comment would be appropriate why is it different specifically for `tron`.

💭 just thinking: we've got codebase full of "if solana, if eth" etc., these kinds coin or network specific stuff should live as pieces of information in static config. But we're long way from there still 🫤

File: packages/suite/src/hooks/wallet/trading/form/useTradingBuyForm.ts [hunk 2]
```diff
@@ -51,12 +50,9 @@ import { useTradingInitializer } from './common/useTradingInitializer';
 import { useTradingFormAccount } from './useTradingFormAccount';
 import { useTradingReceiveAddress } from './useTradingReceiveAddress';
 
-export const useTradingBuyForm = ({
-    pageType = 'form',
```
Comment (Lemonexe) [inline packages/suite/src/hooks/wallet/trading/form/useTradingBuyForm.ts:55]: good catch, `useTradingBuyForm` was never called with anything but the default pageType 👍

**Review by Lemonexe — APPROVED** [review]
Very nice 👌 
Good riddance with all the dead code!
Very nice simplifications in provider selectors, and TradingContainers, and the whole pageType logic chain which is in fact constant 👍
